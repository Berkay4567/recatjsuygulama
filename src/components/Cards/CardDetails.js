import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";

import styles from "./Cards.module.scss";
import { Link } from "react-router-dom";
const CardDetails = () => {
    let { id } = useParams();
  
    let [fetchedData, updateFetchedData] = useState([]);
    let { name, job, avatar, about } = fetchedData;
  
    let api = `https://5fc9346b2af77700165ae514.mockapi.io/simpsons/${id}`;
  
    useEffect(() => {
      (async function () {
        let data = await fetch(api).then((res) => res.json());
        updateFetchedData(data);
      })();
    }, [api]);
  
    return (
      <div className="container d-flex justify-content-center mb-5">
        <div className="d-flex flex-column gap-3">
          <h1 className="text-center">{name}</h1>
          <img className="img-fluid" src={avatar} alt="" />
          <div className="content">
            <div className="">
              <span className="fw-bold">Job : </span>
              {job}
            </div>
            <div className="">
              <span className="fw-bold">About: </span>
              {about}
            </div>
            <br/>
            <Link className="mt-3 w-100 text-center" to="/">
						<button className={styles.button2} >Geri DÃ¶n</button>
			</Link>
          </div>
        </div>
      </div>
    );
  };

export default CardDetails