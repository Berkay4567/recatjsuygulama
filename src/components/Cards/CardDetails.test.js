import React from 'react';
import renderer from 'react-test-renderer'
import { BrowserRouter as Router} from 'react-router-dom';
import CardDetails from './CardDetails';

it("renders when there are no items", ()=>{
    const tree=renderer.create(<Router>
        <CardDetails />
      </Router>
  ).toJSON();
    expect(tree).toMatchSnapshot();
})