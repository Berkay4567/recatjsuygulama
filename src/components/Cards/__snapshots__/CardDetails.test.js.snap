// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no irems 1`] = `
<div
  className="container d-flex justify-content-center mb-5"
>
  <div
    className="d-flex flex-column gap-3"
  >
    <h1
      className="text-center"
    />
    <img
      alt=""
      className="img-fluid"
    />
    <div
      className="content"
    >
      <div
        className=""
      >
        <span
          className="fw-bold"
        >
          Job : 
        </span>
      </div>
      <div
        className=""
      >
        <span
          className="fw-bold"
        >
          About: 
        </span>
      </div>
      <br />
      <a
        className="mt-3 w-100 text-center"
        href="/"
        onClick={[Function]}
      >
        <button
          className="button2"
        >
          Geri Dön
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`renders when there are no items 1`] = `
<div
  className="container d-flex justify-content-center mb-5"
>
  <div
    className="d-flex flex-column gap-3"
  >
    <h1
      className="text-center"
    />
    <img
      alt=""
      className="img-fluid"
    />
    <div
      className="content"
    >
      <div
        className=""
      >
        <span
          className="fw-bold"
        >
          Job : 
        </span>
      </div>
      <div
        className=""
      >
        <span
          className="fw-bold"
        >
          About: 
        </span>
      </div>
      <br />
      <a
        className="mt-3 w-100 text-center"
        href="/"
        onClick={[Function]}
      >
        <button
          className="button2"
        >
          Geri Dön
        </button>
      </a>
    </div>
  </div>
</div>
`;
